Optimizing node for find in files

abose opened the pull request:
- [x] Dont search in files > 16MB 
- [x] only 1 file is indexed at a time instead of 10 in batch.
- [x] File filter has to be applied to indexable files - This will impact the search performance on file filter change[but could be a lesser thig to worry abut], as that will lead to un indexed files.

Tried but not effictive and reverted

```
- [x]  async file reads- this would slow down the indexer considerabily, but might unblock node.
          Reverted to sync file reads as for the brackets project alone, sync file reads complete in ~3 secs instead of 18 secs with async - 6x performance difference. 
- [x] 10 sec delay instead of 5 secs before indexer kick in at first start
         Reverted to 5 secs as this doesn't affect any behavior or solve any problems
```

abose committed:
Dont search in files > 16MB

10 sec delay instead of 5 secs before indexer kick in
only 1 file is indexed at a time instead of 10 in batch.

abose commented:
addresses part of #11613 
I have done some optimizations on it. Will update filter application soon.  
@zaggino Could you check if this pr improves something?

zaggino commented on src/search/node/FindInFilesDomain.js:
```diff
@@ -187,8 +188,26 @@ maxerr: 50, node: true */
+     * @returns {Number} the file size in bytes
+     */
+    function getFilesizeInBytes(fileName) {
+        var stats = {};
```
you don't really need to init with an empty object here, it'll get overridden in try-catch anyway

zaggino commented:
Hi @abose, this is still indexing `node_modules` even though I have them excluded in search filter, which makes Brackets very unresponsive for me. Features that require node connection like `brackets-git` or `brackets-eslint` are literally freezing.

abose commented:
@zaggino Changed the file reads to async. Is there any improvement with this change?

abose committed:
async file crawling.

zaggino commented:
well, actually this doesn't make much difference, probably better to stay with the sync mode ... i found a solution to my problem which is this snippet of code added here: https://github.com/adobe/brackets/blob/master/src/search/FindInFiles.js#L851
```
                console.log('before: files', files.length);
                var filter = FileFilters.getActiveFilter();
                if (filter && filter.patterns.length > 0) {
                    filter = new RegExp(FileFilters.compile(filter.patterns));
                    files = files.filter(function (entry) { return !filter.test(entry); });
                }
                console.log('after: files', files.length);
```
which in my project makes this result:
```
before: files 19649
after: files 397
```
it works well even with the 5 seconds timeout, 10 seconds are not required

abose commented:
Applied file filters before indexing.
I have reverted to sync file reads as for the brackets project alone, sync file reads complete in ~3 secs instead of 18 secs with async - 6x performance difference. Also reverted to 5 sec startup delay.

abose committed:
Applied filters, Reverted to sync file reads as its over 5 times faster to index, and reduced indexing startup time to 5 secs

zaggino committed:
test is about 30% faster than match; also use filterFileList method

zaggino commented:
Looks good to me and solves problems with my projects, hope you don't mind @abose I put a minor tweak commit here, so please have a look. I haven't realized in my code snippet that I can use the filterFileList method which already exists.

One last thing probably, in case user doesn't have his filter configured, we should exclude `bower_components` and `node_modules` here by default, for less handy users, those two folders usually contain tons of files and very rarely someone wants to search them.

abose commented:
Thanks @zaggino for the changes.

prafulVaishnav commented on src/search/FindInFiles.js:
```diff
@@ -840,13 +840,16 @@ define(function (require, exports, module) {
-                    .map(function (entry) {
-                        return entry.fullPath;
-                    });
+                var files = fileListResult,
```
fileListResult is unused.

prafulVaishnav commented:
LGTM..
